coralproject/talk
# Fix/corl 2327 user ban bug

## What does this PR do?
This PR should address https://vmproduct.atlassian.net/browse/CORL-2327 by making user ban management more fine grained.

## What changes to the GraphQL/Database Schema does this PR introduce?
This introduces the `UpdateUserBanInput` input and `UpdateUserBanPayload` type, as well as the `updateUserBan(...)` mutation

## How do I test this PR?
- a "Manage User Ban" modal should now be accessible regardless of a user's status (unblocking site mods from adding additional bans to a user already banned on other sites)
- unscoped moderators should be able to add and remove bans to individual sites for a user (within the same form submission), as well as ban a user across the entire organization, and unban a user across the entire organization
- scoped moderators should be able to add additional single-site bans to users even if that user is already banned on other sites
- scoped mods will see an "All sites" option for banning, which should ban the user from all sites within that moderators scope
- scoped mods will not see a "No Sites" option (as they are not able to remove bans)
- when managing user bans on individual sites, scoped mods should be able to see all sites that a user is banned on, but not be able to unban users on any sites
 
 
## How do we deploy this PR?
I don't think any special considerations should be needed.
